os/board/rtl8730e/src/component/wifi/inic/inic_ipc_host_trx.c

```diff
@@ -217,6 +219,7 @@ void inic_ipc_host_init_skb(void)
 	host_skb_data = (struct skb_data *)rtw_zmalloc(wifi_user_config.skb_num_ap * sizeof(struct skb_data));
 	if (!host_skb_info || !host_skb_data) {
 		DBG_8195A("%s=>skb malloc fail!\n\r", __func__);
+		return;
```

Comment:
Don't you need to free any allocated memory? I mean 
1. both failed -> no need to free
2. alloc host_skb_info, but fail on host_skb_data -> should free host_skb_info
3. fail on info, allocated data -> should free host_skb_data

But this just ends the function. When you call this function again later, previous allocated memory will be a memory leak.

Reply:
Oh beginning of this function checks allocated memory. But that's all. It will fail to do 2nd calling after 1st allocation failure. How about adding freeing allocated one and adding null for init?

Reply:
Thank you for your explanation and suggestion way.
Added the free based on the scenario given.

Reply:
Need to do initialization with NULL.
```
if (info) {
   free(info);
   info = NULL;
}
....
```
Because you check that at beginning of the function.

Reply:
yes, correct... forgot to add the assign to NULL.
Let me add on now.